### PR TITLE
add GitHub Action for Vale

### PR DESCRIPTION
**Pull request by maryamsulemani97**
Part of #1711 

Adds Vale checks to the CI. Copied from the [vale-action](https://github.com/errata-ai/vale-action) repository

### CODE REVIEW

File: .github/workflows/vale.yml
Comment (bidoubiwa) [inline .github/workflows/vale.yml:2]:
```suggestion
on:
  pull_request:
  push:
```

I'm not sure without `pull_request` that it trigger a test whenever you create the PR for the first time

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

File: .github/workflows/vale.yml
Comment (maryamsulemani97) [reply .github/workflows/vale.yml:2]: You're right, thanks! 🙏

File: .github/workflows/vale.yml
Comment (bidoubiwa) [inline .github/workflows/vale.yml:5]:
```suggestion
on:
  pull_request:
  push:
         # trying and staging branches are for BORS config
        branches:
          - main
```

It appears it runs the tests both times since my change. I looked into the other repo's and that is how we do it :)

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥🔥

File: .github/workflows/vale.yml
Comment (bidoubiwa) [inline .github/workflows/vale.yml:5]:
```suggestion
```
my bad!

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥

File: .github/workflows/vale.yml
Comment (curquiza) [inline .github/workflows/vale.yml:6]: I'm pretty sure you don't have any `main` branch but a `master` one 😄 
However I'm not sure these 3 lines are useful: see my global comment of the review

**Review by curquiza — CHANGES_REQUESTED** [review]
Is it something you want as a requirement before merging to `master`?
Because in this case
- you might want to add this check in the `bors.toml` to avoid bors merging your PR if the vale check is not passed
- the 3 lines I commented on are not useful: if you cannot merge a PR without passing this check, no need to run it again on `main`/`master`
- you already have a GHA manifest running on every PR and working with Bors, you might want to update [this file](https://github.com/meilisearch/documentation/blob/master/.github/workflows/check-links.yml) instead of creating a new one

File: bors.toml
Comment (bidoubiwa) [inline bors.toml:4]: I know this is not the point of this PR but you could add the builds in the bors toml as well. 

```
check-builds (14)
check-builds (16)
```

Comment (maryamsulemani97) [reply bors.toml:4]: Does it have to be 
```
check-builds (14)
check-builds (16)
```

Or will just `check-builds` work too?

Comment (bidoubiwa) [reply bors.toml:4]: It has to be as its the names of the jobs



```
check-builds (14)
check-builds (16)
```

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥

**Review by dichotommy — APPROVED** [review]
LGTM! Let's give it a shot 💥

**Comment by maryamsulemani97** [comment]
bors merge

**Comment by maryamsulemani97** [comment]
bors merge

**Pull request merged**
